### PR TITLE
feat: common hud audio design

### PR DESCRIPTION
**Pull request by Landeplage**
# What?
Features and fixes for general HUD audio.

# Why?
Because a HUD is more engaging to use with sound, innit?

### CODE REVIEW

**Review by sandrade-dcl — APPROVED** [review]
Code reviewed and some fixes added.

Only a thing to comment: The Unity tests are taking too long, I don't know if it is due to the status of CI in this moment or there is something on our side that is provoking it.

**Review by menduz — APPROVED** [review]
Approved to remove kernel-team dependency. It doesn't modify any kernel file

File: unity-client/Assets/Tutorial/Scripts/TutorialMusicHandler.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    {
+        if (current)
+        {
+            tutorialHasBeenEnabled = true;
+            TryPlayingMusic();
+        }
+        else
+        {
+            if (tutorialMusic.source.isPlaying)
+                fadeOut = StartCoroutine(tutorialMusic.FadeOut(3f));
```
Comment (BrianAmadori) [inline unity-client/Assets/Tutorial/Scripts/TutorialMusicHandler.cs]:
```suggestion
                fadeOut = CoroutineStarter.Start(tutorialMusic.FadeOut(3f));
```

File: unity-client/Assets/Tutorial/Scripts/TutorialMusicHandler.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        }
+    }
+
+    void TryPlayingMusic()
+    {
+        if (rendererIsReady && tutorialHasBeenEnabled && !tutorialMusic.source.isPlaying)
+        {
+            if (fadeOut != null)
+            {
+                StopCoroutine(fadeOut);
```
Comment (BrianAmadori) [inline unity-client/Assets/Tutorial/Scripts/TutorialMusicHandler.cs]:
```suggestion
                CoroutineStarter.Stop(fadeOut);
```

File: unity-client/Assets/Tutorial/Scripts/TutorialMusicHandler.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            }
+            tutorialMusic.Play();
+        }
+    }
+
+    void OnAvatarEditorMusicPlay()
+    {
+        if (tutorialMusic.source.isPlaying)
+            fadeOut = StartCoroutine(tutorialMusic.FadeOut(1.5f, false));
```
Comment (BrianAmadori) [inline unity-client/Assets/Tutorial/Scripts/TutorialMusicHandler.cs]:
```suggestion
            fadeOut = CoroutineStarter.Start(tutorialMusic.FadeOut(1.5f, false));
```

File: unity-client/Assets/Tutorial/Scripts/TutorialMusicHandler.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    void OnAvatarEditorMusicStop()
+    {
+        if (tutorialMusic.source.isPlaying)
+            StartCoroutine(tutorialMusic.FadeIn(2.5f));
```
Comment (BrianAmadori) [inline unity-client/Assets/Tutorial/Scripts/TutorialMusicHandler.cs]:
```suggestion
            CoroutineStarter.Start(tutorialMusic.FadeIn(2.5f));
```

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Common/Audio/HUDAudioHandler.cs [hunk 5]
```diff
@@ -18,11 +24,35 @@ private void Awake()
...
+        else
+            snapshotOutOfWorld.TransitionTo(0.2f);
+    }
+
+    void OnAllUIHiddenChange(bool current, bool previous)
+    {
+        if (current)
+            AudioScriptableObjects.UIHide.Play(true);
+        else
+            AudioScriptableObjects.UIUnhide.Play(true);
```
Comment (BrianAmadori) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Common/Audio/HUDAudioHandler.cs]:
```suggestion
            AudioScriptableObjects.UIShow.Play(true);
```

**Review by pravusjif — APPROVED** [review]

**Review by pbosio — APPROVED** [review]

File: unity-client/Assets/Tutorial/Scripts/TutorialMusicHandler.cs (hunk 1 above)
Comment (sandrade-dcl) [reply unity-client/Assets/Tutorial/Scripts/TutorialMusicHandler.cs]: Done! Thanks!

File: unity-client/Assets/Tutorial/Scripts/TutorialMusicHandler.cs (hunk 2 above)
Comment (sandrade-dcl) [reply unity-client/Assets/Tutorial/Scripts/TutorialMusicHandler.cs]: Done! Thanks!

File: unity-client/Assets/Tutorial/Scripts/TutorialMusicHandler.cs (hunk 3 above)
Comment (sandrade-dcl) [reply unity-client/Assets/Tutorial/Scripts/TutorialMusicHandler.cs]: Done! Thanks!

File: unity-client/Assets/Tutorial/Scripts/TutorialMusicHandler.cs (hunk 4 above)
Comment (sandrade-dcl) [reply unity-client/Assets/Tutorial/Scripts/TutorialMusicHandler.cs]: Done! Thanks!

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Common/Audio/HUDAudioHandler.cs (hunk 5 above)
Comment (sandrade-dcl) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Common/Audio/HUDAudioHandler.cs]: Done! Thanks!

**Review by BrianAmadori — APPROVED** [review]

**Pull request merged**